### DISCUSSION TITLE
fatfs integration

### DISCUSSION THREAD

**Discussion by satirebird** [micropython/micropython#9618] [General]
I'm trying to integrate micropython to an already existing STM32 project running FreeRTOS. Basically it works but I'm struggle with the vfs. The current firmware uses the fatfs for an sdcard. For my understanding micropython include a modified fatfs. For a new port, the block device interface must be integrated.
Is there a way to mount an already existing fatfs in micropython like in vfs_posix but for fatfs?

**Comment by jimmo** [comment]
@satirebird The reason we use the modified fatfs is because fatfs has global state for a single filesystem. Whereas MicroPython supports mounting multiple concurrent devices (e.g. internal flash and external SDcard).

You could make this work by providing your own version of vfs_fat.c that uses fatfs instead of oofatfs, where you change all the calls to `f_whatever(&self->fatfs)` to just `f_whatever()`.
